src/test/scala/utils/EmbeddedKafka.scala

```diff
@@ -16,39 +16,44 @@ trait EmbeddedKafka[F[_]] {
     zooKeeperPort <- RandomPort[F]
   } yield EmbeddedKafkaConfig(kafkaPort, zooKeeperPort, customBrokerProperties = Map("log.roll.ms" -> "10"))
 
-  def embeddedKafkaR(implicit kafkaConfig: EmbeddedKafkaConfig, F: Async[F]): Resource[F, KafkaServer] =
-    Resource.make(F.delay(Underlying.start().broker))(server => F.delay(server.shutdown()).void)
+  def embeddedKafkaR(kafkaConfig: EmbeddedKafkaConfig)(using F: Async[F]): Resource[F, KafkaServer] =
+    Resource.make(F.delay(Underlying.start()(kafkaConfig).broker))(server => F.delay(server.shutdown()).void)
 
-  def createCustomTopic(topic: String, partitions: Int, topicConfig: Map[String, String])(implicit
+  def createCustomTopic(topic: String, partitions: Int, topicConfig: Map[String, String])(using
       kafkaConfig: EmbeddedKafkaConfig,
       F: Async[F]
   ): F[NonEmptyList[TopicPartition]] =
     for {
-      maybeCreate <-
-        F.delay(Underlying.createCustomTopic(topic = topic, topicConfig = topicConfig, partitions = partitions))
+      tpIdxs      <- F.fromOption(
```

Comment:
minor but this name doesn't mean much to me, whats the x?

Reply:
Just an abbreviation of index

Reply:
[`7064f7f` (#17)](https://github.com/sky-uk/fs2-kafka-topic-loader/pull/17/commits/7064f7ff0ffc1658d6c3ef5c76e652d1eb1f5f8f)

Reply:
ah yeah of course, that didn't register for some reason